src/test/java/iudx/catalogue/server/authenticator/TokenSetup.java

```diff
@@ -0,0 +1,108 @@
+package iudx.catalogue.server.authenticator;
+
+import io.vertx.core.*;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.WebClientOptions;
+import org.jetbrains.annotations.NotNull;
+
+import static iudx.catalogue.server.authenticator.TokensForITs.*;
+
+public class TokenSetup {
+    private static WebClient webClient;
+
+    public static void setupTokens(String authEndpoint, String providerClientId, String providerClientSecret, String consumerClientId, String consumerClientSecret) {
+        // Fetch tokens asynchronously and wait for all completions
+        CompositeFuture.all(
+                fetchToken("provider", authEndpoint, providerClientId, providerClientSecret),
+                fetchToken("admin", authEndpoint, consumerClientId,consumerClientSecret),
+                fetchToken("cosAdmin", authEndpoint, providerClientId, providerClientSecret),
+                fetchToken("consumer", authEndpoint, consumerClientId,consumerClientSecret)
+        ).onComplete(result -> {
+            if (result.succeeded()) {
+                System.out.println("Tokens setup completed successfully");
+            } else {
+                // Handle failure, e.g., log the error
+                System.out.println("errorrrr...");
+                result.cause().printStackTrace();
+            }
+        });
+    }
+
+    private static Future<String> fetchToken(String userType, String authEndpoint, String clientID, String clientSecret) {
+        //System.out.println("user type is..." + userType);
+        Promise<String> promise = Promise.promise();
+        JsonObject jsonPayload = getPayload(userType);
+        //System.out.println("payload: " + jsonPayload);
+
+        // Create a WebClient to make the HTTP request
+        webClient = WebClient.create(Vertx.vertx(), new WebClientOptions().setSsl(true));
+
+        webClient.postAbs(authEndpoint)
+                .putHeader("Content-Type", "application/json")
+                .putHeader("clientID", clientID)
+                .putHeader("clientSecret", clientSecret)
+                .sendJson(jsonPayload)
+                .compose(response -> {
+                    if (response.statusCode() == 200) {
```

Comment:
Also check if response.succeeded before statusCode check to avoid NPE

Reply:
Made the changes that're requested.